File: runtime/botmanager/src/com/newbound/net/service/http/HTTPRequest.java
package com.newbound.net.service.http;

import java.util.Hashtable;

import org.json.JSONObject;

import com.newbound.net.service.Request;

public class HTTPRequest implements Request 
{
	String CMD;
	String LOC;
	String METHOD;
	Hashtable HEADERS;
	Hashtable PARAMS;

	JSONObject log = new JSONObject();
	
	public HTTPRequest(String protocol, String method, String cmd, Hashtable headers, Hashtable params, String loc) 
	{
		try {
			CMD = cmd;
			LOC = loc;
			METHOD = method;
			HEADERS = headers;
			PARAMS = params;

			long now = System.currentTimeMillis();
			log.put("timestamp", now);
			log.put("method", method);
			log.put("path", cmd);
			log.put("protocol", protocol);
			log.put("language", "" + headers.get("ACCEPT-LANGUAGE"));

			String sid = "";
			String c = (String) headers.get("COOKIE");
			if (c != null) {
				String[] cs = c.split("; ");
				for (int i = 0; i < cs.length; i++)
					if (cs[i].startsWith("sessionid=")) {
						sid = cs[i].substring(10);
						headers.put("nn-sessionid", sid);
						break;
					}
			}
			log.put("sessionid", sid);

			String h = (String) headers.get("HOST");
			log.put("HOST", h != null ? h : "");

			String r = (String) headers.get("REFERER");
			log.put("REFERER", r != null ? r : "");
		}
		catch (Exception x) { x.printStackTrace(); }
	}

	@Override
	public Object getCommand() 
	{
		return CMD;
	}

	@Override
	public JSONObject getData() 
	{
		JSONObject jo = new JSONObject();

		try
		{
			jo.put("cmd", CMD);
			jo.put("loc", LOC);
			jo.put("method", METHOD);
			jo.put("headers", new JSONObject(HEADERS));
			jo.put("log", new JSONObject(PARAMS));
		}
		catch (Exception x) { x.printStackTrace(); }

		return jo;
	}

}

File: runtime/botmanager/src/newbound/net/service/http/httprequest.py
from newbound.robot.botutil import BotUtil

class HTTPRequest(BotUtil):

    def __init__(self, protocol, method, cmd, headers, params, loc):
        self.protocol = protocol
        self.method = method
        self.cmd = cmd
        self.headers = headers
        self.params = params
        self.loc = loc

        log = {}
        self.log = log

        now = self.currentTimeMillis()

        log['timestamp'] = now
        log['method'] = method
        log['path'] = cmd
        log['protocol'] = protocol

        language = '*'
        if "ACCEPT-LANGUAGE" in headers:
            language = headers["ACCEPT-LANGUAGE"]
        log['language'] = language

        sid = ''
        if "COOKIE" in headers:
            c = headers['COOKIE']
            cs = c.split("; ")
            for line in cs:
                if line.startswith("sessionid="):
                    sid = line[10:]
                    headers['nn-sessionid'] = sid
                    break
        log['sessionid'] = sid

        host = ''
        if 'HOST' in headers:
            host = headers['HOST']
        log['HOST'] = host

        r = ''
        if 'REFERER' in headers:
            r = headers['REFERER']
        log['REFERER'] = r

        headers['nn-log'] = log

    def getCommand(self):
        return self.cmd

    def getData(self):
        d = {
            'cmd': self.cmd,
            'loc': self.loc,
            'method': self.method,
            'headers': self.headers,
            'log': self.log
        }
        return d
	
	
	
	